recipes/ginkgo/all/conanfile.py

```diff
@@ -139,56 +150,85 @@ def package_info(self):
+        self.cpp_info.components["ginkgo_omp"].set_property(
+            "cmake_target_name", "Ginkgo::ginkgo_omp"
+        )
+        self.cpp_info.components["ginkgo_omp"].libs = ["ginkgo_omp" + debug_suffix]
+        self.cpp_info.components["ginkgo_omp"].requires = ["ginkgo_cuda", "ginkgo_hip"]
 
-        self.cpp_info.components["ginkgo_hip"].set_property("cmake_target_name", "Ginkgo::ginkgo_hip")
-        self.cpp_info.components["ginkgo_hip"].libs = [
-            "ginkgo_hip" + debug_suffix]
+        self.cpp_info.components["ginkgo_hip"].set_property(
+            "cmake_target_name", "Ginkgo::ginkgo_hip"
+        )
+        self.cpp_info.components["ginkgo_hip"].libs = ["ginkgo_hip" + debug_suffix]
 
-        self.cpp_info.components["ginkgo_reference"].set_property("cmake_target_name", "Ginkgo::ginkgo_reference")
+        self.cpp_info.components["ginkgo_reference"].set_property(
+            "cmake_target_name", "Ginkgo::ginkgo_reference"
+        )
         self.cpp_info.components["ginkgo_reference"].libs = [
-            "ginkgo_reference" + debug_suffix]
+            "ginkgo_reference" + debug_suffix
+        ]
 
-        if has_dpcpp_device: # Always add these components
+        if has_dpcpp_device:  # Always add these components
             # See https://github.com/conan-io/conan-center-index/pull/7044#discussion_r698181588
             self.cpp_info.components["ginkgo_core"].requires += ["ginkgo_dpcpp"]
             self.cpp_info.components["ginkgo_core"].requires += ["ginkgo_device"]
 
-            self.cpp_info.components["ginkgo_dpcpp"].set_property("cmake_target_name", "Ginkgo::ginkgo_dpcpp")
+            self.cpp_info.components["ginkgo_dpcpp"].set_property(
+                "cmake_target_name", "Ginkgo::ginkgo_dpcpp"
+            )
             self.cpp_info.components["ginkgo_dpcpp"].libs = [
-                "ginkgo_dpcpp" + debug_suffix]
+                "ginkgo_dpcpp" + debug_suffix
+            ]
 
-            self.cpp_info.components["ginkgo_device"].set_property("cmake_target_name", "Ginkgo::ginkgo_device")
+            self.cpp_info.components["ginkgo_device"].set_property(
+                "cmake_target_name", "Ginkgo::ginkgo_device"
+            )
             self.cpp_info.components["ginkgo_device"].libs = [
-                "ginkgo_device" + debug_suffix]
+                "ginkgo_device" + debug_suffix
+            ]
 
             self.cpp_info.components["ginkgo_omp"].requires += [
-                "ginkgo_dpcpp", "ginkgo_device"]
+                "ginkgo_dpcpp",
+                "ginkgo_device",
+            ]
             self.cpp_info.components["ginkgo_reference"].requires += ["ginkgo_device"]
             self.cpp_info.components["ginkgo_hip"].requires += ["ginkgo_device"]
             self.cpp_info.components["ginkgo_cuda"].requires += ["ginkgo_device"]
             self.cpp_info.components["ginkgo_dpcpp"].requires += ["ginkgo_device"]
+        
+        if has_config_library:
```

Comment:
Just to confirm, this is the only proper change excluding reformatting in this function, right?

Reply:
yes